index.bs

```diff
@@ -2740,30 +4746,104 @@ partial interface MLGraphBuilder {
 
     **Returns:** an {{MLOperand}}. The output 4-D tensor.
 </div>
 
-### The reshape() method ### {#api-mlgraphbuilder-reshape}
+{{MLResample2dOptions}} has the following members:
+<dl dfn-type=dict-member dfn-for=MLResample2dOptions>
+    : <dfn>mode</dfn>
+    ::
+        An {{MLInterpolationMode}} [=string=].
+        Specifies the interpolation algorithm used to fill the output tensor values.
+        The default value is `"nearest-neighbor"`, standing for *Nearest Neighbor* interpolation.
+
+    : <dfn>scales</dfn>
+    ::
+        A sequence of {{float}} of length 2.
+        Specifies the scaling factor in each spatial dimensions of the input: [scaleHeight, scaleWidth].
+        The default value is [1.0, 1.0].
+
+    : <dfn>sizes</dfn>
+    ::
+        A sequence of {{unsigned long}} of length 2.
+        Specifies the target sizes for each spatial dimensions of the input: [sizeHeight, sizeWidth]. When the target sizes are specified, the {{MLResample2dOptions/scales}} argument is ignored, since the scaling factor values are derived from the target sizes of each spatial dimension of the input.
+
+    : <dfn>axes</dfn>
+    ::
+        A sequence of {{unsigned long}} of length 2.
+        Specifies the two consecutive dimensions of the input tensor to which the interpolation algorithm applies. The valid values in the sequence are [0, 1], [1, 2] or [2, 3].
+        The default value is [2, 3].
+</dl>
+
+<details open algorithm>
+
+  <summary>
+    To <dfn for="MLGraphBuilder">check resample options</dfn> given |options|, run the following steps:
+  </summary>
+  <div class=algorithm-steps>
+    1. If |options|.{{MLResample2dOptions/scales}} does not [=map/exist=], set it to to `« 1.0, 1.0 »`.
+    1. Otherwise, if any of its values is not greater than 0, or if its [=list/size=] is not 2, return false.
+    1. If |options|.{{MLResample2dOptions/sizes}} [=map/exists=], and if its size is not 2, or if any of its values is not greater than 0, return false.
+    1. If |options|.{{MLResample2dOptions/axes}} does not [=map/exists=], set it to `« 2, 3 »`.
+    1. Otherwise, if its value is not one of `« 0, 1», « 1, 2», « 2, 3 »`, return false.
+    1. Return true.
+  </div>
+</details>
+
+<details open algorithm>
+
+  <summary>
+    To <dfn for="MLGraphBuilder">resample output sizes</dfn> given |input| and |options|, run the following steps:
+  </summary>
+  <div class=algorithm-steps>
+    1. Let |desc| be an {{MLOperandDescriptor}} initialized to |input|.{{MLOperand/[[descriptor]]}}.
+    1. If |options|.{{MLResample2dOptions/sizes}} [=map/exists=], then set |desc|.{{MLOperand/[[descriptor]]}}.{{MLOperandDescriptor/dimensions}} to |options|.{{MLResample2dOptions/sizes}} and return |desc|.
```

Comment:
|options|.{{MLResample2dOptions/sizes}}  only indicated the output sizes in two dimensions specified by {{MLResample2dOptions/axes}}